voidcrew/code/game/machinery/cassette/advanced_cassette_deck.dm

```diff
@@ -0,0 +1,152 @@
+/obj/machinery/cassette/adv_cassette_deck
+	name = "Advanced Cassette Deck"
+	desc = "A more advanced less portable Cassette Deck. Useful for recording songs from our generation, or customizing the style of your cassettes."
+	icon = 'voidcrew/icons/obj/machines/adv_cassette_deck.dmi'
+	icon_state = "cassette_deck"
+	density = TRUE
+	///cassette tape used in adding songs or customizing
+	var/obj/item/device/cassette_tape/tape
+	///Selection used to add the jukebox as a song to a cassette
+	var/datum/track/selection = null
+
+/obj/machinery/cassette/adv_cassette_deck/attackby(obj/item/object, mob/user, params)
+	if(!(flags_1 & NODECONSTRUCT_1))
+		if(object.tool_behaviour == TOOL_WRENCH)
+			if(!anchored && !isinspace())
+				to_chat(user,"<span class='notice'>You secure [src] to the floor.</span>")
+				set_anchored(TRUE)
+			else if(anchored)
+				to_chat(user,"<span class='notice'>You unsecure and disconnect [src].</span>")
+				set_anchored(FALSE)
+			playsound(src, 'sound/items/deconstruct.ogg', 50, TRUE)
+			return
+	return ..()
+
+/obj/machinery/cassette/adv_cassette_deck/attackby(obj/item/cassette, mob/user)
```

Comment:
Theres two of these... but its fine because the other one can be converted to `wrench_act`